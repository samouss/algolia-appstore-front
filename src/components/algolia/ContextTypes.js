import PropTypes from 'prop-types';

export default {
  algoliaClient: PropTypes.object.isRequired,
  algoliaHelper: PropTypes.object.isRequired,
};
