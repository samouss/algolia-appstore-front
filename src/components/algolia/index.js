export { default as Provider } from './Provider';
export { default as withPaginateResults } from './withPaginateResults';
