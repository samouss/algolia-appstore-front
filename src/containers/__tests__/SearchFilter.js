import React from 'react';
import { shallow } from 'enzyme';
import { createMockAlgoliaClient, createMockAlgoliaHelper } from 'test/algolia';
import WithQuery, { SearchFilter } from '../SearchFilter';

describe('<SearchFilter />', () => {
  const defaultProps = {
    query: 'This is a query',
    onChange: () => {},
  };

  it('expect to render', () => {
    const props = {
      ...defaultProps,
    };

    const component = shallow(
      <SearchFilter
        {...props}
      />,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to call onChange', () => {
    const props = {
      ...defaultProps,
      onChange: jest.fn(),
    };

    const component = shallow(
      <SearchFilter
        {...props}
      />,
    );

    component
      .find('TextInput')
      .simulate('change', {
        currentTarget: {
          value: 'query',
        },
      });

    expect(props.onChange).toHaveBeenCalledWith('query');
  });

  describe('withQuery', () => {
    it('expect to render', () => {
      const context = {
        algoliaClient: createMockAlgoliaClient(),
        algoliaHelper: createMockAlgoliaHelper(),
      };

      const component = shallow(
        <WithQuery />,
        { context },
      );

      expect(component).toMatchSnapshot();
    });
  });
});
