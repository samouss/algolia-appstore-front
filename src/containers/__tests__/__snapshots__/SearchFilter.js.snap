// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchFilter /> expect to render 1`] = `
<TextInput
  enableClearButton={true}
  onChange={[Function]}
  placeholder="Angry Birds, Cut the Rope, Pinterest..."
  value="This is a query"
/>
`;

exports[`<SearchFilter /> withQuery expect to render 1`] = `
<SearchFilter
  onChange={[Function]}
  query=""
/>
`;
